stubs/tensorflow/tensorflow/keras/callbacks.pyi

```diff
@@ -0,0 +1,188 @@
+    def on_train_end(self, logs: _Logs = None) -> None: ...
+
+# A CallbackList has exact same api as a callback,
+# but does not actually subclass it.
+class CallbackList:
+    def __init__(
+        self,
+        callbacks: Sequence[Callback] | None = None,
+        add_history: bool = False,
+        add_progbar: bool = False,
+        # model: Model[Any, object] | None = None,
+        model: Model | None = None,
+        **params: Any,
+    ) -> None: ...
+    # def set_model(self, model: Model[Any, object]) -> None: ...
+    def set_model(self, model: Model) -> None: ...
+    def set_params(self, params: dict[str, Any]) -> None: ...
+    def on_batch_begin(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_batch_end(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_epoch_begin(self, epoch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_epoch_end(self, epoch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_predict_batch_begin(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_predict_batch_end(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_predict_begin(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_predict_end(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_test_batch_begin(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_test_batch_end(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_test_begin(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_test_end(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_train_batch_begin(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_train_batch_end(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_train_begin(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_train_end(self, logs: Mapping[str, object] | None = None) -> None: ...
+
+class History(Callback):
+    history: dict[str, list[Any]]
+
+class ModelCheckpoint(Callback):
+    monitor_op: Any
+    filepath: str
+    _options: tf.train.CheckpointOptions | tf.saved_model.SaveOptions | None
+    def __init__(
+        self,
+        filepath: str,
+        monitor: str = "val_loss",
+        verbose: Literal[0, 1] = 0,
+        save_best_only: bool = False,
+        save_weights_only: bool = False,
+        mode: Literal["auto", "min", "max"] = "auto",
+        save_freq: str | int = "epoch",
+        options: tf.train.CheckpointOptions | tf.saved_model.SaveOptions | None = None,
+        initial_value_threshold: float | None = None,
+    ) -> None: ...
+    def _save_model(self, epoch: int, batch: int | None, logs: _Logs) -> None: ...
+
+class TensorBoard(Callback):
```

Comment:
Minor Nit: Can we have defined callbacks in alphabetical order?

Reply:
Done in 8fef1b215